store/models/job_spec.go

```diff
@@ -236,17 +245,59 @@ func (t TaskSpec) MarshalJSON() ([]byte, error) {
 	return json.Marshal(merged)
 }
 
+// TaskType defines what Adapter a TaskSpec will use.
+type TaskType string
+
+// NewTaskType returns a formatted Task type.
+func NewTaskType(val string) (TaskType, error) {
+	re := regexp.MustCompile("^[a-zA-Z0-9-_]*$")
+	if !re.MatchString(val) {
+		return TaskType(""), fmt.Errorf("Task Type validation: name %v contains invalid characters", val)
+	}
+
+	return TaskType(strings.ToLower(val)), nil
+}
+
+// MustNewTaskType instantiates a new TaskType, and panics if a bad input is provided.
+func MustNewTaskType(val string) TaskType {
+	tt, err := NewTaskType(val)
+	if err != nil {
+		panic(fmt.Sprintf("%v is not a valid TaskType", val))
```

Comment:
Can we change this to `logger.Panic`?

Reply:
We can't because the logger package requires the models package, so it becomes a circular dependency. I'm up to refactor it, but seems out of scope for this PR.